test/acceptance/display_matching_invocations_alongside_expectations_test.rb

```diff
@@ -56,4 +56,23 @@ def test_should_display_yields_of_matching_invocations_alongside_expectations_in
       '  - #<Mock:foo>.bar(3, 4) # => raised StandardError after yielding ("b", "c"), then ("d", "e")'
     ], test_result.failure_message_lines
   end
+
+  def test_should_display_empty_yield_and_return
```

Comment:
Minor: I assume you've added this test just for completeness/coverage and not because it identified a problem with the implementation, i.e. I assume it would have passed even if you'd added it earlier. Is that correct? If so, it would be good to amend the earliest possible commit to include it. If on the other hand this test drove out some implementation changes, it would be good to include it in the commit where those changes were made.

Reply:
> I tried inserting this into the right earlier commit using fixup but the rebase was too painful, so I gave up. D'you have a suggestion for how I could achieve that without having to redo all subsequent commits?

Not without trying it myself. In this kind of situation I often create a new branch and cherry-pick individual commits onto it. Then when I know a given commit is going to cause conflicts I  create a *new* commit with the intention of the commit in the original branch. If the commits are all relatively small, this isn't too painful. However, I realise this can be quite time-consuming.

Reply:
Resolved in https://github.com/freerange/mocha/pull/394#issuecomment-551927383.